src/backend/distributed/utils/listutils.c

```diff
@@ -59,6 +59,11 @@ SortList(List *pointerList, int (*comparisonFunction)(const void *, const void *
 
 	pfree(array);
 
+	if (sortedList != NIL)
```

Comment:
I'm surprised that we haven't set this for a long time. I guess there are no other such fields that we need to set?